### DISCUSSION TITLE
Subscription not getting canceled

### DISCUSSION THREAD

**Discussion by esfomeado** [enisdenjo/graphql-sse#52] [Q&A]
I'm using this library with Apollo and after I unsubscribe from the subscription the connection isn't closed. The server doesn't receive any message at all when that happens.

It works fine via Postman or Altair. Is there anything else that needs to be done to cancel/complete the subscription?

**Comment by enisdenjo** [comment]
"With Apollo" is very broad and vague. What is your server setup? Do you use graphql-sse on the server too? What does your client usage look like? A repro would be very helpful.

**Reply by esfomeado** [reply]
I will try to give you a repo next week or so.

On the server I'm using DGS implementation and it works fine with other clients.

On the client side I followed your example and created a link and then I just use the Apollo client as I would normally would to do a request and them at the end I unsubscribe.

I gave a look at the Altair implementation that uses this library and they use the client directly and at the end they call the dispose method and it works fine with my server.

So should I use the grahpql-sse client directly instead of wrapping it in a link and do a request with the Apollo client?

**Reply by enisdenjo** [reply]
Beware that graphql-sse implements the [GraphQL over SSE spec](https://github.com/enisdenjo/graphql-sse/blob/master/PROTOCOL.md), and the server must too in order to work properly. 

> So should I use the grahpql-sse client directly instead of wrapping it in a link and do a request with the Apollo client?

Would be great to try, it will bring us more insight on where the issue might be.
